storage_controller/src/tenant_shard.rs

```diff
@@ -1352,6 +1357,7 @@ impl TenantShard {
             }) => {
                 // It's legal to remove a secondary that is not present in the intent state
                 !self.intent.secondary.contains(&new_node_id)
+                    && self.intent.attached != Some(new_node_id)
```

Comment:
I find the comment confusing, even before your PR.
Is it referring to the line below it or is justifying the absence of a check like `self.intent.secondary.contains(&old_node_id)` ?
Probalby @VladLazar can answer.

In any way, an additional comment on the clause that you added here would make sense, I think?
Like:
```
// new_node_id became the intent attached location since we planned the optimization
```